internal/document/editor/editorservice/service.go

```diff
@@ -70,8 +71,13 @@ func (s *parserServiceServer) Serialize(_ context.Context, req *parserv1.Seriali
 
 	cells := make([]*editor.Cell, 0, len(req.Notebook.Cells))
 	for _, cell := range req.Notebook.Cells {
-		if _, ok := cell.Metadata["id"]; !ok && cell.Kind == parserv1.CellKind_CELL_KIND_CODE {
-			cell.Metadata["id"] = idgen.GenerateID()
+		if req.Identity.Number() == parserv1.RunmeIdentity_RUNME_IDENTITY_ALL.Number() ||
```

Comment:
I'm fairly certain you can compare the props without calling `Number()` on it, no?